### DISCUSSION TITLE
How to use RSA 1024 key client credentials to make requests

### DISCUSSION THREAD

**Discussion by vazmin** [hoppscotch/hoppscotch#4547] [Q&A] [answered]
I can't change the key for now. The higher version of openssl does not support RSA 1024 Key by default.

When using the local `curl` with RSA 1024, the following error occurs:  
```
curl: (58) could not load PEM client certificate, OpenSSL error error:0A00018F:SSL routines::ee key too small, (no key found, wrong pass phrase, or wrong file format?)
```

However, I can bypass this error by specifying the security level:  
```
[system_default_sect]
CipherString = DEFAULT@SECLEVEL=1
```

When using **Hoppscotch Desktop**, the error is:  
`Problem with the local SSL certificate`

When using **Hoppscotch Agent**, the error is:  
`Internal Server Error`

How can I configure either the **Desktop** or **Agent** to lower the OpenSSL security level?

**Comment by SanskritiHarmukh** [comment, marked as answer]
Heyy, after checking, it looks like our platform enforces OpenSSL's default security level (SECLEVEL=2), which blocks RSA-1024 keys. Unfortunately, there’s no way to lower the security level directly on Hoppscotch.

There are some advanced workarounds like using a reverse proxy or bridge, but those can get pretty complex. The easiest and most secure option would be to upgrade your RSA key to a stronger one, like RSA-2048 or higher, to meet the security requirements. Really sorry for the inconvenience.

**Reply by vazmin** [reply]
Got it, thanks a lot.
